std/algorithm/comparison.d

```diff
@@ -1567,3 +1567,156 @@ unittest
                 2, "two",
                 ));
 }
+
+/** Returns: First Argument (element of $(D a)) whose implicit conversion to
+    bool is true.
+
+    Similar to behaviour of or operator in dynamic languages such as Lisp's (or
+    a...) and Python's a or ....
+
+    TODO Is inout Conversion!T the correct return value?
+
+    NOTE: Lazy parameters are currently marked as throw which makes it
+    impossible to mark either() as nothrow. Issue at
+    https://issues.dlang.org/show_bug.cgi?id=12647
+*/
+CommonType!Ts either(Ts...)(lazy Ts a) if (a.length >= 1)
+{
+    auto a0 = a[0]();           // evaluate only once
+    static if (Ts.length == 1)
+    {
+        return a0;
+    }
+    else
+    {
+        return a0 ? a0 : either(a[1 .. $]); // recurse
```

Comment:
There's no need for this to be recursive, and it will definitely be slower than an iterative approach.